### PR TITLE
[MOB-9995] reorganize source code

### PR DESCRIPTION
**Pull request by lposen**
## 🔹 JIRA Ticket(s) if any

* [MOB-9995](https://iterable.atlassian.net/browse/MOB-9995)

## ✏️ Description

Reorganizes source code for better readability
Organizes source code into `core`, `inApp` and `inbox` folders -- eventually, these should be made into packages to reduce bloat in client apps.

[MOB-9995]: https://iterable.atlassian.net/browse/MOB-9995?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/inApp/enums/IterableInAppTriggerType.ts [hunk 1]
```diff
@@ -7,5 +7,3 @@ export enum IterableInAppTriggerType {
   /** Do not display the in-app automatically via the SDK */
   never = 2,
 }
-
-export default IterableInAppTriggerType;
```
Comment (evantk91) [inline src/inApp/enums/IterableInAppTriggerType.ts]: Ah I see. Disregard the comments on this. :)

**Review by evantk91 — APPROVED** [review]
Looking great! Thoroughly looked through all the imports and ran the example apps.

File: src/inbox/IterableInbox.tsx [hunk 2]
```diff
@@ -11,20 +11,19 @@ import {
...
-import IterableInboxMessageList from './IterableInboxMessageList';
-import useAppStateListener from './useAppStateListener';
-import useDeviceOrientation from './useDeviceOrientation';
+} from '../IterableInAppClasses';
+import type { IterableInboxCustomizations } from '../IterableInboxCustomizations';
+import IterableInboxDataModel from '../IterableInboxDataModel';
+import IterableInboxEmptyState from '../IterableInboxEmptyState';
+import IterableInboxMessageDisplay from '../IterableInboxMessageDisplay';
+import IterableInboxMessageList from '../IterableInboxMessageList';
+import { useAppStateListener, useDeviceOrientation } from '../hooks';
```
[the remark this answers is not in the document]
Comment (lposen) [reply src/inbox/IterableInbox.tsx]: Thanks!

File: src/Iterable.ts [hunk 3]
```diff
@@ -12,12 +12,12 @@ import {
...
+  IterableInAppManager,
+  IterableInAppMessage,
   IterableInAppCloseSource,
   IterableInAppDeleteSource,
   IterableInAppLocation,
-} from './IterableInAppClasses';
-import IterableInAppManager from './IterableInAppManager';
-import IterableInAppMessage from './IterableInAppMessage';
+} from './inApp';
 import { IterableLogger } from './IterableLogger';
```
[the remark this answers is not in the document]
Comment (lposen) [reply src/Iterable.ts:23]: Great!  I'm glad

File: src/inApp/IterableInAppClasses.ts [hunk 4]
```diff
@@ -1,132 +0,0 @@
...
-    icon: string | undefined
-  ) {
-    this.title = title;
-    this.subtitle = subtitle;
-    this.icon = icon;
-  }
-
-  static fromDict(dict: any): IterableInboxMetadata {
-    return new IterableInboxMetadata(dict.title, dict.subtitle, dict.icon);
-  }
```
[the remark this answers is not in the document]
Comment (lposen) [reply src/inApp/IterableInAppClasses.ts]: Thanks!

File: src/Iterable.ts [hunk 5]
```diff
@@ -19,64 +19,16 @@ import {
...
 } from './inApp';
 import { IterableLogger } from './IterableLogger';
+import type IterableCommerceItem from './IterableCommerceItem';
 
 const RNIterableAPI = NativeModules.RNIterableAPI;
 const RNEventEmitter = new NativeEventEmitter(RNIterableAPI);
 
-enum AuthResponseCallback {
+enum IterableAuthResponseResult {
   SUCCESS,
```
[the remark this answers is not in the document]
Comment (lposen) [reply src/Iterable.ts]: Thanks!  Trying to make sure that it isn't confusing for the client when importing things from our code.

File: src/IterableConfig.ts [hunk 6]
```diff
@@ -1,13 +1,11 @@
...
 import type { IterableActionContext } from './IterableActionContext';
+import type { IterableAuthResponse } from './IterableAuthResponse';
 import { IterableDataRegion } from './IterableDataRegion';
 import { IterablePushPlatform } from './IterablePushPlatform';
 import { IterableLogLevel } from './enums';
 import { IterableInAppMessage, IterableInAppShowResponse } from './inApp';
 
-// TODO: Add description
-type AuthCallBack = () => void;
-
```
[the remark this answers is not in the document]
Comment (lposen) [reply src/IterableConfig.ts:13]: It's such a simple type that it's not really needed

**Pull request merged**